ed25519_test.go

```diff
@@ -56,6 +64,34 @@ func TestEd25519Verify(t *testing.T) {
 		if !data.valid && err == nil {
 			t.Errorf("[%v] Invalid key passed validation", data.name)
 		}
+
+		// test key rotations
+		invalidKey, err := jwt.ParseEdPublicKeyFromPEM([]byte(wrongPubKeyData))
+		if err != nil {
+			t.Errorf("Unable to parse wrong Ed25519 public key: %v", err)
+		}
+
+		err = method.Verify(strings.Join(parts[0:2], "."), parts[2], []ed25519.PublicKey{})
+		if err == nil {
+			t.Errorf("[%v] Empty keys passed validation", data.name)
+		}
+
+		err = method.Verify(strings.Join(parts[0:2], "."), parts[2], []ed25519.PublicKey{invalidKey.(ed25519.PublicKey)})
+		if err == nil {
+			t.Errorf("[%v] Invalid keys passed validation", data.name)
+		}
+
+		if !data.valid {
+			continue
+		}
+
+		err = method.Verify(strings.Join(parts[0:2], "."), parts[2], []ed25519.PublicKey{
+			invalidKey.(ed25519.PublicKey),
+			ed25519Key.(ed25519.PublicKey),
```

Comment:
NOTE: the typecastings here are needed because [`ParseEdPublicKeyFromPEM`](https://pkg.go.dev/github.com/golang-jwt/jwt/v4#ParseEdPublicKeyFromPEM) returns `crypto.PublicKey` instead of `ed25519.PublicKey`, which, in my opinion, was a big mistake (all other similar functions, for example `ParseRSAPublicKeyFromPEM`, returns the expected type).

I can see this could become a footgun, as people might try to return `[]crypto.PublicKey` in their `Keyfunc`. If we think that's a problem we want to prevent, there are two ways to do that:

1. Make `SigningMethodEd25519.Verify` to also accept `[]crypto.PublicKey`, but this will make that one more special than others;
2. Deprecate `ParseEdPublicKeyFromPEM` and use a different named function to return `ed25519.PublicKey`. If we go this route we probably would want to do the same with `ParseEdPrivateKeyFromPEM`.

(We can't just fix `ParseEdPublicKeyFromPEM` because that's a breaking change, unfortunately)

Reply:
done in [new commit](https://github.com/golang-jwt/jwt/pull/170/commits/21672254b6120ffc4fd9d008c83f01b49631163f)

Reply:
Thanks for putting that through, overall this looks good to me given the available options.